compiler/rustc_hir_typeck/src/check.rs

```diff
@@ -266,7 +266,7 @@ fn check_lang_start_fn<'tcx>(
         let fn_generic = generics.param_at(0, tcx);
         let generic_ty = tcx.mk_ty_param(fn_generic.index, fn_generic.name);
         let expected_fn_sig =
-            tcx.mk_fn_sig([].iter(), &generic_ty, false, hir::Unsafety::Normal, Abi::Rust);
+            tcx.mk_fn_sig([].into_iter(), generic_ty, false, hir::Unsafety::Normal, Abi::Rust);
```

Comment:
I wonder if we can use `ty::List::empty()` here?

Reply:
I tried it, it doesn't work because `ty::List::empty()` is not an iterator.

The underlying problem here is that `mk_fn_sig` unusually *doesn't* have an `intern_fn_sig` alternative that takes a slice. That's because it combines the iterator with the return value before interning, so `intern_fn_sig` would have to do additional work to assembler the slice `[args, ..., retval]` anyway.

Reply:
Oh, but once I change `mk_fn_sig` to use `I: IntoIterator`, then this can just be `mk_fn_sig([], ...)`!